hazelcast/src/main/java/com/hazelcast/datalink/impl/DataLinkServiceImpl.java

```diff
@@ -72,19 +71,19 @@ private void processDataLinkRegistrations(ClassLoader classLoader) {
 
     @Override
     public void createConfigDataLink(DataLinkConfig config) {
-        put(config, CONFIG, true);
+        put(config, CONFIG);
     }
 
-    private void put(DataLinkConfig config, DataLinkSource source, boolean replace) {
+    private void put(DataLinkConfig config, DataLinkSource source) {
         dataLinks.compute(config.getName(),
                 (key, current) -> {
                     if (current != null) {
-                        if (!replace) {
-                            throw new HazelcastException("Data link '" + config.getName() + "' already exists");
-                        }
                         if (current.source == CONFIG) {
                             throw new HazelcastException("Cannot replace a data link created from configuration");
                         }
+                        if (current.config.equals(config)) {
+                            return current;
+                        }
                         // close the old DataLink
                         try {
                             current.instance.release();
```

Comment:
if the `release()` method takes long time (can involve network communication) then entire operation may be hanging and also subsequent updates of the datalink will be waiting. Shouldn't it be done in async way?